### PR TITLE
feat(text-input): SPIKE

### PR DESCRIPTION
**Pull request by andresin87**
## feat(text-input): Spike

**TASK**: 

### Description, Motivation and Context

I have followed the same pattern as checkbox and radio input API:
```
<InputElement>
  labelfortahtelement
<InputElement>
```
means:
```
<label>
  <input/>
</label>
```

- TextInput: easy add inputs
- Input: html-input and its span

First attempt to manage inputs from spark

### Types of changes

- [x] ✨ New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: packages/components/text-input/src/Input.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+export const Input = forwardRef<HTMLInputElement, InputProps>(
+  (
+    { className, mandatory, htmlSize, intent, placeholder, children, onValueChange, ...props },
+    forwardedRef
+  ) => {
+    const [value, setValue] = useCombinedState(props.value, props.defaultValue)
+    const onChangeHandler = (event: ChangeEvent<HTMLInputElement>) => {
+      const value = event.target?.value?.slice(0, htmlSize)
+      setValue(value)
+      typeof onValueChange === 'function' && onValueChange(value)
```
Comment (andresz1) [inline packages/components/text-input/src/Input.tsx:55]: Maybe would be better to call `onValueChange` with the event 🤔 ?

File: packages/components/text-input/src/InputField.styles.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    'border-md',
+    'border-current',
+    'items-center',
+    // 'overflow-hidden'
+  ],
+  {
+    variants: {
+      /**
+       * Color scheme of the inputField.
+       */
```
Comment (andresz1) [inline packages/components/text-input/src/InputField.styles.tsx:19]: Is this comment needed ?

File: packages/components/text-input/src/InputField.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+export const InputField = forwardRef<HTMLElement, InputFieldProps>(
+  ({ asChild, intent, disabled, className, children, ...props }, forwardedRef) => {
+    const Element = asChild ? Slot : 'span'
+
+    return (
+      <Element
+        ref={forwardedRef}
+        aria-disabled={disabled}
```
Comment (andresz1) [inline packages/components/text-input/src/InputField.tsx:21]: Is this needed ?

File: packages/components/text-input/src/Input.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+export const Input = forwardRef<HTMLInputElement, InputProps>(
+  (
+    { className, mandatory, htmlSize, intent, placeholder, children, onValueChange, ...props },
+    forwardedRef
+  ) => {
+    const [value, setValue] = useCombinedState(props.value, props.defaultValue)
+    const onChangeHandler = (event: ChangeEvent<HTMLInputElement>) => {
+      const value = event.target?.value?.slice(0, htmlSize)
```
[the remark this answers is not in the document]
Comment (andresin87) [reply packages/components/text-input/src/Input.tsx:53]: the default size of an input is 25ch width unless you configure an htmlSize which means you determine how much characters should input had (html size native attribute)

File: packages/components/text-input/src/Input.tsx (hunk 1 above)
Comment (andresin87) [reply packages/components/text-input/src/Input.tsx:53]: I'm following what we have in checkbox and radio for the onValueChange event handler responses

[the remark this answers is not in the document]
Comment (andresin87) [reply packages/components/text-input/src/Input.tsx:53]: it's just for if it exists, you mean...

```suggestion
     onValueChange && onValueChange(value)
```
just for checking if its `!== undefined` cuz the type is handled by Ty?

**Pull request closed without merging**